wsaa.py

```diff
@@ -136,42 +138,62 @@ def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
-            )
-        finally:
-            # Liberar memoria asignada
-            _lib.BIO_free(bio_in)
-        # Se crea un buffer nuevo porque la firma lo consume
-        bio_in = _lib.BIO_new_mem_buf(tra, len(tra))
-        try:
-            # Crear buffer de salida
-            bio_out = _lib.BIO_new(_lib.BIO_s_mem())
+        if sys.version.split(" ")[0][0:3] == "2.7":
             try:
-                # Instanciar un SMIME
-                _lib.SMIME_write_PKCS7(bio_out, p7, bio_in, 0)
-
-                # Tomar datos para la salida
-                result_buffer = _ffi.new("char**")
-                buffer_length = _lib.BIO_get_mem_data(bio_out, result_buffer)
-                output = _ffi.buffer(result_buffer[0], buffer_length)[:]
+                # Firmar el texto (tra) usando cryptography (openssl bindings para python)
+                p7 = _lib.PKCS7_sign(
+                    cert._x509, private_key._evp_pkey, _ffi.NULL, bio_in, 0
+                )
+            finally:
+                # Liberar memoria asignada
+                _lib.BIO_free(bio_in)
+            # Se crea un buffer nuevo porque la firma lo consume
+            bio_in = _lib.BIO_new_mem_buf(tra, len(tra))
+            try:
+                # Crear buffer de salida
+                bio_out = _lib.BIO_new(_lib.BIO_s_mem())
+                try:
+                    # Instanciar un SMIME
+                    _lib.SMIME_write_PKCS7(bio_out, p7, bio_in, 0)
+
+                    # Tomar datos para la salida
+                    result_buffer = _ffi.new("char**")
+                    buffer_length = _lib.BIO_get_mem_data(bio_out, result_buffer)
+                    output = _ffi.buffer(result_buffer[0], buffer_length)[:]
+                finally:
+                    _lib.BIO_free(bio_out)
             finally:
-                _lib.BIO_free(bio_out)
-        finally:
-            _lib.BIO_free(bio_in)
+                _lib.BIO_free(bio_in)
+
+            # Generar p7 en formato mail y recortar headers
+            msg = email.message_from_string(output.decode("utf8"))
+            for part in msg.walk():
+                filename = part.get_filename()
+                if filename == "smime.p7m":
+                    # Es la parte firmada?
+                    # Devolver CMS
+                    return part.get_payload(decode=False)
+
+        p7 = pkcs7.PKCS7SignatureBuilder().set_data(
+            tra
+        ).add_signer(
+            cert, private_key, hashes.SHA256()
+        ).sign(
+            serialization.Encoding.SMIME, [pkcs7.PKCS7Options.Binary]
+        )
 
         # Generar p7 en formato mail y recortar headers
-        msg = email.message_from_string(output.decode("utf8"))
+        msg = email.message_from_string(p7.decode("utf8"))
         for part in msg.walk():
             filename = part.get_filename()
-            if filename == "smime.p7m":
+            if filename == "smime.p7s":
```

Comment:
One possibility to avoid the `m` and `s` problem is to use the `startswith` method.
For instance:
```suggestion
            if filename and filename.startswith("smime.p7"):
```